### ISSUE TITLE
gtk-3.x: bugzilla patch breaks compilation

### ISSUE THREAD

**Issue by rb2k** [NixOS/nixpkgs#133787]
### Describe the bug

It seems like [the patch included in the gtk derivation](https://github.com/NixOS/nixpkgs/blob/6fc5211eddddc02c50ca7f98d6cc377726417fa9/pkgs/development/libraries/gtk/3.x.nix#L82) does now return an xz files, but fetchpatch does not understand that:

```
builder for '/nix/store/9cv52mcrm864khwrx3yvvkpm5cpjnsqw-Xft-setting-fallback-compute-DPI-properly.patch.drv' failed with exit code 1; last 10 log lines:
  trying https://bug757142.bugzilla-attachments.gnome.org/attachment.cgi?id=344123
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  100   169  100   169    0     0    290      0 --:--:-- --:--:-- --:--:--   290
  100   888  100   888    0     0    833      0  0:00:01  0:00:01 --:--:--   833
  error: Normalized patch '/private/tmp/nix-build-Xft-setting-fallback-compute-DPI-properly.patch.drv-0/0g6fhqcv8spfy3mfmxpyji93k8d4p4q4fz1v9a1c1cgcwkz41d7p' is empty (while the fetched file was not)!
  Did you maybe fetch a HTML representation of a patch instead of a raw patch?
  Fetched file was:
  �7zXZ���ִF�!����X��N5]�#����q�9���]E�͌o���%$�����]�Ht�Hަ/p�ѵ����d�{��;1e���Lx�m���-�ћ�!��v�r�)Q���<��g%�������8�;�()�%*��Rf�F�<��0�䍒���H^ 
  [...]
  ```

Following the URL, it does indeed specify the content type and name with .xz:

```
% curl -X GET -I "https://bug757142.bugzilla-attachments.gnome.org/attachment.cgi?id=344123"
HTTP/1.1 301 Moved Permanently
Server: nginx/1.20.1
Date: Fri, 13 Aug 2021 16:42:32 GMT
Content-Type: text/html
Content-Length: 169
Location: https://bugzilla.gnome.org/attachment.cgi?id=344123
[...]

% curl -X GET -I "https://bugzilla.gnome.org/attachment.cgi?id=344123"
HTTP/1.1 200 OK
Server: nginx/1.20.1
Date: Fri, 13 Aug 2021 16:42:47 GMT
Content-Type: application/x-xz
Content-Length: 888
Last-Modified: Tue, 27 Jul 2021 06:49:26 GMT
ETag: "253748d050c620c4d9be6f4e19bd85a6"
Cache-Control: max-age=31536000
X-Cache-Status: HIT
Content-Disposition: attachment; filename="attachment_344123.xz"
Accept-Ranges: bytes
Set-Cookie: 7d27d72c49a5c89ddd41aaf4d7ac81d7=1d0fd3cfb313d48f5fa934ea2d63afab; path=/; HttpOnly; Secure; SameSite=None
Cache-control: private
```

### Steps To Reproduce
1. try to build gtk-3.x
2. watch it fail

### Expected behavior
Compile should succeeed

### Notify maintainers
@raskin
@hedning
@jtojnar
@dasj19
@maxeaubrey

**Comment by rb2k** [comment]
fixed with #133818

**Comment by jtojnar** [comment]
Same issue with gnome-common, [caribou](https://gitlab.gnome.org/GNOME/caribou/commit/13df8b92ae89c796238e669ee6ef4447a42d6355.patch), cogl and gtk2.
